src/main/java/codeu/controller/ProfilePagesServlet.java

```diff
@@ -32,12 +47,59 @@ void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
 
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
   /**
    * This function fires when a user goes to the profile pages.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-        request.getRequestDispatcher("/WEB-INF/view/profile-pages.jsp").forward(request, response);
+    String requestUrl = request.getRequestURI();
+    String username = requestUrl.substring(USERNAME_INDEX);
+
+    User user = userStore.getUser(username);
+    UUID userId = user.getId();
```

Comment:
You can probably just combine this into 
`UUID userID = userStore.getUser(username).getId();`

It's fairly obvious what the intermediate variables are. I only suggest declaring intermediate variables when code is confusing without them or they are needed elsewhere.